### PR TITLE
ContaoFramework must be able to boot with a request without route

### PR DESCRIPTION
**Pull request by aschempp**

### CODE REVIEW

File: src/Framework/ContaoFramework.php [hunk 1]
```diff
@@ -243,10 +243,14 @@ private function getRoute()
...
-            $this->request->attributes->get('_route'),
-            $this->request->attributes->get('_route_params')
-        );
+        try {
+            $route = $this->router->generate(
+                $this->request->attributes->get('_route'),
+                $this->request->attributes->get('_route_params')
+            );
+        } catch (\InvalidArgumentException $e) {
+            return null;
```
Comment (discordier) [inline src/Framework/ContaoFramework.php]: PHP Doc for return value must be changed to `@return string|null` also.

Comment (aschempp) [inline src/Framework/ContaoFramework.php]: question is if `null` is correct? Or if `console` should be `null` too?

Comment (leofeyer) [inline src/Framework/ContaoFramework.php]: I guess it should be `null` on the command line, too, then.

Comment (aschempp) [inline src/Framework/ContaoFramework.php]: I agree, but should that be considered a BC break? I very much doubt anyone uses `TL_SCRIPT` on the command line, as it's only necessary for the legacy `initialize.php`…

Comment (leofeyer) [inline src/Framework/ContaoFramework.php]: I also doubt that anyone uses `TL_SCRIPT` an the command line, but if you are worried about BC, we can just keep the return value.

**Comment by leofeyer** [comment]
Merged in a0c5d6da39b498b1eff4a0df44d32995caa99cf9.

**Pull request merged**